src/write_sensu.c

```diff
@@ -39,6 +39,53 @@
 #include <stddef.h>
 
 #include <stdlib.h>
+#ifdef HAVE_KERNEL_SOLARIS
```

Comment:
Unfortunately not as there is a function declaration in the header file. Undefining applies only to the preprocessor and a C declaration cannot be easily "undeclared". So my solution change the declaration by a macro definition to another name and then clears that renaming so the substitute goes in with the right name.

Another solution would be to use something like my_asprintf in the code and then make definitions to that to either the substitute or the builtin.